### PR TITLE
Trace common disk operations

### PR DESCRIPTION
**Pull request by tempoz**
Adds tracing to common file operations using the disk utility. Avoids changing function signatures.

---

**Version bump**: Minor

### CODE REVIEW

File: server/util/disk/disk.go [hunk 1]
```diff
@@ -76,12 +83,19 @@ func FileExists(fullPath string) (bool, error) {
...
-type SectionReaderCloser struct {
+type readCloser struct {
 	*io.SectionReader
 	io.Closer
+	ctx context.Context
 }
 
-func FileReader(ctx context.Context, fullPath string, offset, length int64) (*SectionReaderCloser, error) {
+func (r *readCloser) Read(p []byte) (int, error) {
+	_, spn := tracing.StartSpan(r.ctx)
```
Comment (tylerwilliams) [inline server/util/disk/disk.go:93]: ctx, spn := tracing.StartSpan(r.ctx)

File: server/util/disk/disk.go [hunk 2]
```diff
@@ -91,16 +105,23 @@ func FileReader(ctx context.Context, fullPath string, offset, length int64) (*Se
...
 }
 
 type writeMover struct {
 	*os.File
+	ctx       context.Context
 	finalPath string
 }
 
+func (w *writeMover) Write(p []byte) (int, error) {
+	_, spn := tracing.StartSpan(w.ctx)
```
Comment (tylerwilliams) [inline server/util/disk/disk.go:120]: ctx, spn := tracing.StartSpan(w.ctx)

**Review by tylerwilliams — CHANGES_REQUESTED** [review]
See inline comments

**Review by vadimberezniker — APPROVED** [review]

File: server/util/disk/disk.go (hunk 1 above)
Comment (tempoz) [reply server/util/disk/disk.go:93]: So when I went to do this, I realized there is no context variable referenced in this function, which means I can't name the variable. Same for below.

Comment (tylerwilliams) [reply server/util/disk/disk.go:93]: lol i'm a goof, thanks for looking :)

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**